7_react/react1/week3/todo-app/src/component/TodoContainer.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react';
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import 'bootstrap/dist/css/bootstrap.min.css'
+import {FaCalendarAlt} from 'react-icons/fa';
+
+//function for calender icon
+function CustomInput({value,onClick}){
+    return(
+        <div className='input-group'>
+            <input type='text' className='form-control' value={value} onClick={onClick} readOnly/>
+            <div className='input-group-append'>
+                <span className='input-group-text'>
+                    <FaCalendarAlt/>
+                </span>
+            </div>
+        </div>
+    )
+}
+
+const TodoContainer = ({ jsonTodos }) => {
+  const [newTodo, setNewTodo] = useState('');
+  const [todos, setTodos] = useState([]);
+  const [selectedDate, setDate] = useState(new Date());
+
+  useEffect(() => {
+    setTodos(jsonTodos);
```

Comment:
I think you can move the whole fetching logic here :) There is no need to have useEffect 2x doing the same thing